src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +33,17 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        AtomicInteger index = new AtomicInteger();
+        return numbers.stream()
+                .map((n) -> {
+                    n -= index.get() % 2 != 0 ? 1 : 0;
+                    index.getAndIncrement();
+                    return n;
+                })
+                .filter(n -> n % 2 != 0)
+                .mapToDouble(n -> n)
+                .average()
+                .orElseThrow(NoSuchElementException::new);
```

Comment:
If go this way we reduce the amount of operations
```suggestion
                .average()
                .getAsDouble();
```